### PR TITLE
[Spark-7949] [MLlib] [Doc] update document with some missing save/load

### PR DESCRIPTION
**Pull request by hhbyyh**
add save load for examples:
KMeansModel
PowerIterationClusteringModel
Word2VecModel
IsotonicRegressionModel

### CODE REVIEW

**Comment by jkbradley** [comment]
@hhbyyh  I don't see any problems.  Just to confirm, have you run all of the updated examples?

**Comment by hhbyyh** [comment]
@jkbradley. Sure, I have verified the changes.

File: docs/mllib-clustering.md [hunk 1]
```diff
@@ -63,6 +63,10 @@ val clusters = KMeans.train(parsedData, numClusters, numIterations)
 val WSSSE = clusters.computeCost(parsedData)
 println("Within Set Sum of Squared Errors = " + WSSSE)
 {% endhighlight %}
+
+// Save and load model
```
Comment (jkbradley) [inline docs/mllib-clustering.md]: Oops!  I missed this before.  This should go before the endhighlight.

Also, you now need to import KMeansModel

File: docs/mllib-clustering.md [hunk 2]
```diff
@@ -143,6 +151,10 @@ def error(point):
 
 WSSSE = parsedData.map(lambda point: error(point)).reduce(lambda x, y: x + y)
 print("Within Set Sum of Squared Error = " + str(WSSSE))
+
+# Save and load model
+clusters.save(sc, "myModelPath")
+sameModel = KMeansModel.load(sc, "myModelPath")
```
Comment (jkbradley) [inline docs/mllib-clustering.md]: ditto: import KMeansModel

File: docs/mllib-clustering.md [hunk 3]
```diff
@@ -325,6 +337,10 @@ val model = pic.run(similarities)
 model.assignments.foreach { a =>
   println(s"${a.id} -> ${a.cluster}")
 }
+
+// Save and load model
+model.save(sc, "myModelPath")
+val sameModel = PowerIterationClusteringModel.load(sc, "myModelPath")
```
Comment (jkbradley) [inline docs/mllib-clustering.md]: ditto: import PowerIterationClusteringModel

File: docs/mllib-feature-extraction.md [hunk 4]
```diff
@@ -201,6 +201,11 @@ val synonyms = model.findSynonyms("china", 40)
 for((synonym, cosineSimilarity) <- synonyms) {
   println(s"$synonym $cosineSimilarity")
 }
+
+// Save and load model
+model.save(sc, "myModelPath")
+val sameModel = Word2VecModel.load(sc, "myModelPath")
```
Comment (jkbradley) [inline docs/mllib-feature-extraction.md]: import Word2VecModel

File: docs/mllib-isotonic-regression.md [hunk 5]
```diff
@@ -88,6 +88,10 @@ val predictionAndLabel = test.map { point =>
 // Calculate mean squared error between predicted and real labels.
 val meanSquaredError = predictionAndLabel.map{case(p, l) => math.pow((p - l), 2)}.mean()
 println("Mean Squared Error = " + meanSquaredError)
+
+// Save and load model
+model.save(sc, "myModelPath")
+val sameModel = IsotonicRegressionModel.load(sc, "myModelPath")
```
Comment (jkbradley) [inline docs/mllib-isotonic-regression.md]: import IsotonicRegressionModel

**Comment by jkbradley** [comment]
I missed these items on my first pass---and I remember making the same mistakes when I first wrote save/load examples!

**Comment by hhbyyh** [comment]
This is awkward... Thanks a lot for the careful review.
Guess I just focused on the save/load part.

**Comment by jkbradley** [comment]
No problem (and thanks for catching that other typo I missed).  Merging into master and branch-1.4

**Pull request closed without merging**